graph_altimeter/scan/__init__.py

```diff
@@ -88,32 +89,28 @@ def run(config, resource_specs=None):  # pylint: disable=too-many-locals
         scanned_accounts,
         not_scanned_accounts,
     )
+    neptune_client = GraphAltimeterNeptuneClient(endpoint)
 
+    logger.debug('scan %s: writing results to Gremlin DB', scan_id)
     graph_set = AltimeterUniverseGraph(graph_set)
-    graph, snapshot_vertex_id = graph_set.to_gremlin_lpg(scan_id)
-
-    neptune_client = AltimeterNeptuneClient(
-        max_age_min=1440,
-        neptune_endpoint=endpoint,
-    )
-
-    # TODO: The following two operations are not transactional. This should be
-    # reviewed.
-    logger.debug('scan %s: writing results in Gremlin DB', scan_id)
-
-    neptune_client.write_to_neptune_lpg(graph, scan_id)
-    link_snapshot_with_universe(neptune_client, snapshot_vertex_id)
-    logger.debug('scan %s: finished writting results in Gremlin DB', scan_id)
-
-
-def link_snapshot_with_universe(neptune_client, vertex_id):
-    """Links the current universe to a vertex given its vertex_id. It also
-    ensures that the current ``Universe`` vertex exists."""
+    graph = graph_set.to_gremlin_lpg(scan_id)
+    snapshot_vertex = add_snapshot_vertex(neptune_client, scan_id)
+    neptune_client.write_to_neptune_lpg(graph, scan_id, snapshot_vertex)
+    logger.debug('scan %s: finished writting results to Gremlin DB', scan_id)
+
+
+def add_snapshot_vertex(neptune_client, scan_id):
+    """Creates a Altimeter snapshot vertex and links it to the proper universe. It
+    also ensures that the current ``Universe`` vertex exists. Returns
+    the vertex of the new created vertex."""
+    # The vertex_id of the snapshot must follow a concrete naming convention.
+    vertex_id = f"altimeter_snapshot_{scan_id}"
     _, conn = neptune_client.connect_to_gremlin()
     g = traversal(AltimeterTraversalSource).withRemote(conn)
     g.ensure_universe(CURRENT_UNIVERSE).next()
-    g.link_to_universe(CURRENT_UNIVERSE, vertex_id).next()
+    vertex = g.add_snapshot(vertex_id, CURRENT_UNIVERSE).next()
     conn.close()
```

Comment:
Should we use try-except-finally here to guarantee that the connection is properly closed?